cocoa/tests_backend/widgets/button.py

```diff
@@ -0,0 +1,36 @@
+from pytest import xfail
+
+from toga_cocoa.libs import NSBezelStyle, NSButton
+
+from .base import SimpleProbe
+from .properties import toga_color, toga_font
+
+
+class ButtonProbe(SimpleProbe):
+    native_class = NSButton
+
+    @property
+    def text(self):
+        return self.native.title
+
+    @property
+    def color(self):
+        xfail("Can't get/set the text color of a button on macOS")
+
+    @property
+    def font(self):
+        return toga_font(self.native.font)
+
+    @property
+    def background_color(self):
+        if self.native.bezelColor:
+            return toga_color(self.native.bezelColor)
+        else:
+            return None
+
+    def assert_display_properties(self):
+        # If the button is tall (for any reason), it should have a different bezel style.
+        if self.native.frame.size.height > 50:
+            assert self.native.bezelStyle == NSBezelStyle.RegularSquare
+        else:
+            assert self.native.bezelStyle == NSBezelStyle.Rounded
```

Comment:
I don't particularly like having a standalone assertion class either (especially given it will be empty on most implementations) - but having assertions on an attribute probe bugs me as well. I think it makes a certain amount of sense to maintain a separation between "things being measured" and "things being asserted". 

That said, I think you may be right. Completely aside from the issue of having mostly empty implementations, for Button on Cocoa, we need to assert that when the button's height changes, the properties of the button also change. That obviously happens when a test changes the size of the button, but it also happens implicitly if a test changes the font - something that the current font change property tests doesn't check for.

So - as a sort of "general policy" statement, I guess it makes sense to say that if an implementation-specific property (like `bezelStyle`) is a direct consequence of an interface-level property (like `height`), we should include an assertion in the probe's attribute access. 

If I'm reading the slider example correctly, the same is true there as well. There's an interface level probe property reporting the number of ticks, which returns `None` if the widget doesn't have ticks; but there's also a specific implementation requirement that if the widget hasn't been configured for tick marks, `numberOfTickMarks` must be 0. I'm not sure if this is because of some API issue where having `numberOfTickMarks != 0` and `allowsTickMarkValuesOnly == False` causes a GUI issue, or it's just a case of keeping a clean house internally - but either way, it's a native level concern stemming from an observable interface property, and changes to the interface property should impact on the native rendition of that property.